misc.c

```diff
@@ -2069,9 +2069,11 @@ GC_API void GC_CALL GC_enable(void)
     LOCK();
     GC_ASSERT(GC_dont_gc != 0); /* ensure no counter underflow */
     GC_dont_gc--;
+#ifndef BUFFERED_FINALIZATION
```

Comment:
Why is this `#ifndef`ed?

Reply:
When we allocate the pages for the finaliser buffer, we do so from within the STW pause which can cause the heap to grow if lots of objects are enqueued for finalisation in a single GC. This `WARN` prints to stderr anytime the heap grows, and in alloy tests this can cause spurious output which causes a test to fail because stderr does not match.

I probably should have made this a separate commit, happy for me to do so when I squash?

Reply:
> I probably should have made this a separate commit, happy for me to do so when I squash?

Works for me!